contracts/finance/andromeda-set-amount-splitter/src/contract.rs

```diff
@@ -0,0 +1,315 @@
+use crate::state::SPLITTER;
+use andromeda_finance::set_amount_splitter::{
+    validate_recipient_list, AddressAmount, ExecuteMsg, GetSplitterConfigResponse, InstantiateMsg,
+    QueryMsg, Splitter,
+};
+use andromeda_std::{
+    ado_base::{InstantiateMsg as BaseInstantiateMsg, MigrateMsg},
+    amp::messages::AMPPkt,
+    common::{actions::call_action, encode_binary, Milliseconds, MillisecondsDuration},
+    error::ContractError,
+};
+use andromeda_std::{ado_contract::ADOContract, common::context::ExecuteContext};
+use cosmwasm_std::{
+    attr, coins, ensure, entry_point, BankMsg, Binary, Coin, CosmosMsg, Deps, DepsMut, Env,
+    MessageInfo, Reply, Response, StdError, SubMsg,
+};
+use cw_utils::nonpayable;
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:andromeda-set-amount-splitter";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+// 1 day in seconds
+const ONE_DAY: u64 = 86_400;
+// 1 year in seconds
+const ONE_YEAR: u64 = 31_536_000;
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    let splitter = match msg.lock_time {
+        Some(ref lock_time) => {
+            // New lock time can't be too short
+            ensure!(
+                lock_time.get_time(&env.block).seconds() >= ONE_DAY,
+                ContractError::LockTimeTooShort {}
+            );
+
+            // New lock time can't be too long
+            ensure!(
+                lock_time.get_time(&env.block).seconds() <= ONE_YEAR,
+                ContractError::LockTimeTooLong {}
+            );
+            Splitter {
+                recipients: msg.recipients.clone(),
+                lock: lock_time.get_time(&env.block),
+            }
+        }
+        None => {
+            Splitter {
+                recipients: msg.recipients.clone(),
+                // If locking isn't desired upon instantiation, it's automatically set to 0
+                lock: Milliseconds::default(),
```

Comment:
I think `lock` should be consistent. According to the validation process when `lock_time` is given, it is `ONE_DAY` is used as minimum and `ONE_YEAR` is used as maximum. But here, zero value is assigned to `lock`. If `ONE_DAY` is supposed to be lower limit, `lock` should be assigned with a value that is equal or greater than `ONE_DAY`. I suggest using one day as a default value

Reply:
In that case the lock would no longer be optional. 
There's no right or wrong here, but my thinking is why reduce a user's options. The lock is optional and if it's desired, should at least be for one day.